File: mobile/src/main/java/ro/antiprotv/sugar/db/Category.java
package ro.antiprotv.sugar.db;

public enum Category {
    FOOD(1), DRINKS(2), CLEANING(3), OTHER(99);
    int code;
    Category(int code) {
        this.code = code;
    }

    public static int getCodeFromString(String category) {
        int code = 99;
        for (Category c : Category.values()) {
            if (c.toString().equals(category)) {
                code = c.code;
                break;
            }
        }
        return code;
    }

    public static Category fromString(String category) {
        Category cat = Category.OTHER;
        for (Category c : Category.values()) {
            if (c.toString().equals(category)) {
                cat = c;
                break;
            }
        }
        return cat;
    }

    public static String fromCode(int code) {
        Category cat = Category.OTHER;
        for (Category c : Category.values()) {
            if (c.code == code) {
                cat = c;
                break;
            }
        }
        return cat.toString();

    }
}

File: mobile/src/main/java/ro/antiprotv/sugar/repository/db/Category.kt
package ro.antiprotv.sugar.repository.db

enum class Category(val code: Int, val resId: String) {
    FOOD(1, "item_category_food"),
    DRINK(2, "item_category_drink"),
    CLEANING(3, "item_category_cleaning"),
    OTHER(99, "item_category_other");

    companion object {
        fun fromString(category: String): Category {
            var cat = OTHER
            for (c in values()) {
                if (c.toString() == category) {
                    cat = c
                    break
                }
            }
            return cat
        }

        fun fromCode(code: Int): Category {
            var cat = OTHER
            for (c in values()) {
                if (c.code == code) {
                    cat = c
                    break
                }
            }
            return cat
        }
    }

}